### PR TITLE
add distributed incident creation rate limiting

### PR DESCRIPTION
**Pull request by diegocepedaw**
if a plan breaches the quota set by incident_quota_max
        it will no longer be able to create incidents. After
        quota_breach_duration amount of seconds from the first incident
        the plan's quota will reset.

### CODE REVIEW

File: src/iris/cache.py [hunk 1]
```diff
@@ -92,7 +97,58 @@ def cache_modes():
...
+        it will no longer be able to create incidents. After
+        quota_breach_duration amount of seconds from the first incident
+        the plan's quota will reset.
+    '''
+    global quota_configs
+    if quota_configs.get('disable_quota'):
+        # skip quota check
+        return False
+    try:
+        mc = pylibmc.Client([quota_configs.get('memcache_host')], binary=True, behaviors={"cas": True, "tcp_nodelay": True, "ketama": True})
```
Comment (colinxy) [inline src/iris/cache.py]: You are creating a new memcached client every time `check_quota_breach` is called. This code is in the hot path of incident creation, so it's probably a good idea to use connection pooling.

File: src/iris/cache.py [hunk 2]
```diff
@@ -92,7 +97,55 @@ def cache_modes():
...
```
Comment (colinxy) [inline src/iris/cache.py]: You are creating a new memcached client every time check_quota_breach is called. This code is in the hot path of incident creation, so it's probably a good idea to use connection pooling.

Comment (diegocepedaw) [reply src/iris/cache.py]: Good point, I'll add that

**Comment by diegocepedaw** [comment]
repaced with NGINX rate rimiting

**Pull request closed without merging**